src/query/storages/fuse/src/operations/mutation/update/update_source.rs

```diff
@@ -0,0 +1,383 @@
+                    let progress_values = ProgressValues {
+                        rows: data_block.num_rows(),
+                        bytes: 0,
+                    };
+                    self.scan_progress.incr(&progress_values);
+                    self.state = State::UpdateData(data_block);
+                }
+            }
+            State::MergeRemain {
+                part,
+                chunks,
+                mut data_block,
+                filter,
+            } => {
+                if let Some(remain_reader) = self.remain_reader.as_ref() {
+                    let remain_block = remain_reader.deserialize_parquet_chunks(part, chunks)?;
+                    for (col, field) in remain_block
+                        .columns()
+                        .iter()
+                        .zip(remain_block.schema().fields())
+                    {
+                        data_block = data_block.add_column(col.clone(), field.clone())?;
+                    }
+                }
+
+                let field = DataField::new("_predicate", bool::to_data_type());
+                data_block = data_block.add_column(filter, field)?;
+
+                self.state = State::UpdateData(data_block);
+            }
+            State::UpdateData(data_block) => {
+                let func_ctx = self.ctx.try_get_function_context()?;
+                let block = self
+                    .operators
+                    .iter()
+                    .try_fold(data_block, |input, op| op.execute(&func_ctx, input))?;
+                self.state = State::NeedSerialize(block);
+            }
+            State::NeedSerialize(block) => {
+                let row_count = block.num_rows() as u64;
+                let block_size = block.memory_size() as u64;
+                let (block_location, block_id) = self.location_gen.gen_block_location();
+
+                // build block index.
+                let location = self.location_gen.block_bloom_index_location(&block_id);
+                let (bloom_index_state, column_distinct_count) =
+                    BloomIndexState::try_create(&block, location)?;
+                let col_stats = gen_columns_statistics(&block, Some(column_distinct_count))?;
+
+                // serialize data block.
+                let mut block_data = Vec::with_capacity(100 * 1024 * 1024);
+                let schema = block.schema().clone();
+                let (file_size, meta_data) = blocks_to_parquet(
+                    &schema,
+                    vec![block],
+                    &mut block_data,
+                    self.table_compression,
+                )?;
+                let col_metas = util::column_metas(&meta_data)?;
+
+                // new block meta.
+                let new_meta = Arc::new(BlockMeta::new(
+                    row_count,
+                    block_size,
+                    file_size,
+                    col_stats,
+                    col_metas,
+                    None,
```

Comment:
do we need to update the clustering stats here? or there will be a dedicated PR to update the clustering stats?

Reply:
> update the clustering stats

https://github.com/datafuselabs/databend/issues/9616